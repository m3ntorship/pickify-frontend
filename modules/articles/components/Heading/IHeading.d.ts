declare namespace IHeading {
  export interface IProps {
    text: string;
    toBeRed?: boolean;
  }
}
export { IHeading };
