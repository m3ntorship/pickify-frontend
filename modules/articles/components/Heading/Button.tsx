import * as React from 'react';

export function giveMeFive(): number {
  const five = 5;
  return five;
}

interface Props {
  buttonText: string;
}

export default ({ buttonText }: Props): React.ReactElement => (
  <button
    type="button"
    onClick={(): void => {
      console.log(giveMeFive());
    }}
  >
    {buttonText}
  </button>
);
