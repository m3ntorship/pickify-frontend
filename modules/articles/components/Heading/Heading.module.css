.btn {
  @apply font-bold;
}
.text_red {
  color: red;
  @apply text-primary;
}
