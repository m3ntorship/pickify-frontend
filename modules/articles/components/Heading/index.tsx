import * as React from 'react';
import classNames from 'classnames';
import type { FC, ReactElement } from 'react';
import styles from './Heading.module.css';
import type { IHeading } from './IHeading';

const Heading: FC<IHeading.IProps> = ({ text, toBeRed }): ReactElement => {
  const h2Class: string = classNames({
    [styles.text_red]: toBeRed,
  });
  return (
    <h2 className={h2Class} data-testid="h2-element">
      {' '}
      {text}
      <p className={styles.btn}>sdsads</p>
      <p className="text-8xl">ali</p>
    </h2>
  );
};
export default Heading;
