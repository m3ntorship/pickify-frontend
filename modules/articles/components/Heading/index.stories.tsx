import type { ReactElement } from 'react';
import Heading from '.';

export default {
  component: Heading,
  title: 'TS/Heading',
};

export const HeadingTS = (): ReactElement => (
  <Heading text="political articles" />
);
