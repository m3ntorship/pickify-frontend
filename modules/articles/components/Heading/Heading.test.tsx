import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import type { TargetElement } from '@testing-library/user-event';
import Heading from './index';

describe('example of a test suite', () => {
  it('example of single test', () => {
    const tree = renderer.create(<Heading text="hi" />);
    expect(tree).toMatchSnapshot();
  });

  it('h2 should have color red', () => {
    const { getByTestId } = render(<Heading text="hi" toBeRed />);

    const h2Element: TargetElement = getByTestId('h2-element');

    expect(h2Element).toHaveClass('text_red');
  });
});
