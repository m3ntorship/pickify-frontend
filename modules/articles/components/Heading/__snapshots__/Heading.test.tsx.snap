// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example of a test suite example of single test 1`] = `
<h2
  className=""
  data-testid="h2-element"
>
   
  hi
  <p
    className="btn"
  >
    sdsads
  </p>
  <p
    className="text-8xl"
  >
    ali
  </p>
</h2>
`;
